src/transformers/configuration_utils.py

```diff
@@ -75,9 +75,9 @@ def __init__(self, **kwargs):
         self.top_k = kwargs.pop("top_k", 50)
         self.top_p = kwargs.pop("top_p", 1.0)
         self.repetition_penalty = kwargs.pop("repetition_penalty", 1.0)
-        self.bos_token_id = kwargs.pop("bos_token_id", 0)
-        self.pad_token_id = kwargs.pop("pad_token_id", 0)
-        self.eos_token_ids = kwargs.pop("eos_token_ids", 0)
+        self.bos_token_id = kwargs.pop("bos_token_id", None)
+        self.pad_token_id = kwargs.pop("pad_token_id", None)
+        self.eos_token_ids = kwargs.pop("eos_token_ids", None)
```

Comment:
cc @thomwolf what do you think? Now models that do not have those tokens won't fail silently as they did previously, as `None` can't be used in pytorch/tf tensors.

Reply:
I like this!

Reply:
A problem with setting `self.bos_token_id = None` as a default value and allowing it in the assert statements of the `generate` function in modeling_utils.py is that we will get an error when the `input_ids` argument of the `generate` function is set set to `None`. In this case `torch.full(...)` in line 732 in modeling_utils.py will throw an error (see below).